src/canvas-effects/canvas-effect.js

```diff
@@ -3392,46 +3115,57 @@ export default class CanvasEffect extends PIXI.Container {
 				(creationTimeDifference % (loopDuration * 1000)) / 1000;
 		}
 
-		return this._resetLoop();
+		if (this._loopOffset) {
+			this.mediaCurrentTime = this._loopOffset
+		}
+		await this.playMedia();
+		this._addToTicker(this.loopHandler)
 	}
 
-	/**
-	 * Continuously reset the video to the right time so that the start and end time can be preserved
-	 *
-	 * @returns {Promise<void>}
-	 * @private
-	 */
-	async _resetLoop(firstLoop = true) {
-		if (this._ended) return;
-
-		let loopWaitTime = 0;
-		if (this._animationTimes.loopStart) {
-			if (this._isEnding) return;
-			this.mediaCurrentTime = (firstLoop ? 0 : this._animationTimes.loopStart) + (this._loopOffset > 0 ? this._loopOffset : 0);
-			loopWaitTime = (this._animationTimes.loopEnd - this.mediaCurrentTime) * 1000;
-		} else {
-			this.mediaCurrentTime = this._startTime + this._loopOffset;
-			loopWaitTime = this._animationDuration - this._loopOffset * 1000;
+	async loopHandler() {
+		if (this._ended || this.isEnding) {
+			return;
+		}
+		if (this.mediaCurrentTime < this._endTime) {
+			return;
+		}
+		if (this.restartLoopHandler != null) {
+			return;
 		}
 
-		await this.playMedia();
+		// if we're above end time, we can safely just pause for now
+		this.pauseMedia()
```

Comment:
Do we want to await here? Not sure.

Reply:
Yes i think so. Playing is an asynchronous operation in the DOM API. 
maybe we could get away with not awaiting here but I don't really like to think about all the asynchronous interactions and race conditions that could have...